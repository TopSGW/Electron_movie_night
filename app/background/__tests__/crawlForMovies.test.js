'use strict'
/* globals describe, test, expect, jest */
const path = require('path')

// This is the module we are testing
const { crawlForMovies } = require('../crawlForMovies')

const crawlParams = {
  rootDirectory: __dirname,
  searchDirCb: jest.fn(),
  movieFileCb: jest.fn()
}

describe('crawlForMovies', () => {
  test('discovers test movies without crashing', () => {
    return expect(crawlForMovies(crawlParams)).resolves.toBeUndefined()
  })

  test('calls search directory callback during crawl', () => {
    const testDirs = crawlParams.searchDirCb.mock.calls.map(dir => {
      const { name } = path.parse(dir[0])
      return name
    })

    expect(testDirs).toMatchSnapshot()
  })

  test('calls movie file callback during crawl', () => {
    const movies = crawlParams.movieFileCb.mock.calls.map(file => {
      const { name, ext } = path.parse(file[0])
      return `${name}${ext}`
    })

    expect(movies).toMatchSnapshot()
  })
})
