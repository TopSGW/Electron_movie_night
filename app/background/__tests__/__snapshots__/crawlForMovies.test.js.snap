// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crawlForMovies calls movie file callback during crawl 1`] = `
Array [
  "2012 (2009).mov",
  "I Dont Know How She Does It (2011).webm",
  "Killers (2010).m4v",
  "Old School.avi",
  "Star Trek 2009.mp4",
  "TheDuchess[2008].mpeg",
  "sample.avi",
  "test footage.mp4",
]
`;

exports[`crawlForMovies calls search directory callback during crawl 1`] = `
Array [
  "__tests__",
  "__fixtures__",
  "empty_movie_files_for_testing",
  "__snapshots__",
]
`;
